stochastic_physics/stochastic_physics_wrapper.F90

```diff
@@ -317,61 +304,37 @@ subroutine stochastic_physics_wrapper (GFS_Control, GFS_Data, Atm_block, ierr)
              ca_turb_cpl(nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%ca_turb(:)
              ca_shal_cpl(nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%ca_shal(:)
          enddo
-         call cellular_automata_sgs(GFS_Control%kdt,GFS_control%dtf,GFS_control%restart,GFS_Control%first_time_step,              &
-            sst,lmsk,lake,condition,ca_deep_cpl,ca_turb_cpl,ca_shal_cpl,ca_deep_diag,ca_turb_diag,                                &
-            ca_shal_diag,Atm(mygrid)%domain_for_coupler,nblks,                                                                    &
+         call cellular_automata_sgs(GFS_Control%kdt,GFS_control%dtp,GFS_control%restart,GFS_Control%first_time_step,              &
+            sst,lmsk,lake,condition,ca_deep_cpl,ca_turb_cpl,ca_shal_cpl, Atm(mygrid)%domain_for_coupler,nblks,                    &
             Atm_block%isc,Atm_block%iec,Atm_block%jsc,Atm_block%jec,Atm(mygrid)%npx,Atm(mygrid)%npy, levs,                        &
-            GFS_Control%nthresh,GFS_Control%rcell,GFS_Control%nca,GFS_Control%scells,GFS_Control%tlives,GFS_Control%nfracseed,    &
-            GFS_Control%nseed,GFS_Control%ca_global,GFS_Control%ca_sgs,GFS_Control%iseed_ca,                                      &
-            GFS_Control%ca_smooth,GFS_Control%nspinup,GFS_Control%ca_trigger,Atm_block%blksz(1),GFS_Control%master,GFS_Control%communicator)
+            GFS_Control%nthresh,GFS_Control%rcell,GFS_Control%tile_num,GFS_Control%nca,GFS_Control%scells,GFS_Control%tlives,     &
+            GFS_Control%nfracseed, GFS_Control%nseed,GFS_Control%iseed_ca,                                                        &
+            GFS_Control%nspinup,GFS_Control%ca_trigger,Atm_block%blksz(1),GFS_Control%master,GFS_Control%communicator)
          ! Copy contiguous data back as needed
          do nb=1,nblks
-             GFS_Data(nb)%Intdiag%ca_deep(:)  = ca_deep_diag(nb,1:GFS_Control%blksz(nb))
-             GFS_Data(nb)%Intdiag%ca_turb(:)  = ca_turb_diag(nb,1:GFS_Control%blksz(nb))
-             GFS_Data(nb)%Intdiag%ca_shal(:)  = ca_shal_diag(nb,1:GFS_Control%blksz(nb))
+             GFS_Data(nb)%Intdiag%ca_deep(:)  = ca_deep_cpl(nb,1:GFS_Control%blksz(nb))
```

Comment:
Wondering, why do we need to copy `ca_deep_cpl` into both `GFS_Data(nb)%Intdiag%ca_deep` and `GFS_Data(nb)%Coupling%ca_deep`? Do we need to have two copies? Same for `ca_shal` and `ca_turb`.

Reply:
We have that so we can output the ca fields using the diag_table.  Is not no longer necessary?

Reply:
It's based on rather old code, from 2017, at the time the fields in GFS_diagnostics.F90 needed to be allocated in the "Diag" container, whereas Coupling fields were in the "Coupling" container defined in GFS_typedefs.F90. They are the same fields, just that one is coupled to the physics and one is used for diagnostics output as defined by diag_table. Perhaps there is a better way of doing this now?

Reply:
I just realized that I didn't address this code change.  Just so you know, this is not an increase in the memory footprint of the code compared to what is in develop. I just moved to copies out of the CA code.

Reply:
That's ok, I think there are a few more of those duplicate arrays that we should clean up in the near future.

Reply:
I just did a clean up of the ca and sppt,shum,skeb etc. duplicates.  If it works, I will commit it.